docs/src/resolution_independence.md

```diff
@@ -0,0 +1,74 @@
+
+The shape of the physical pixel could be complex and definitely varies from display model to model.
+However for simplicity you can think of a pixel as a square which you can choose a color for.
+
+## Display pixel density
+
+As technology advances the physical size of pixels is getting smaller and smaller.
+This allows display manufacturers to put more and more pixels into the same sized screen.
+The **pixel densities of displays are increasing**.
+
+There is also an **increasing variety in the pixel density** of the displays used by people.
+Some might have a brand new *30" 8K UHD* (*7680px \* 4320px*) display,
+while others might still be rocking their *30" HD ready* (*1366px \* 768px*) display.
+It might even be the same person on the same computer with a multi-display setup.
+
+## The naive old school approach to UI
+
+For a very long time UIs have been designed without thinking about pixel density at all.
+People tended to have displays with roughly similar pixel densities, so it all kind of
+worked most of the time. However **it breaks down horribly** in a modern world.
+The *200px \* 200px* UI that looks decent on that *HD ready* display is barely visible
+on the *8K UHD* display. If you redesign it according to the *8K UHD* display then
+it won't even fit on the *HD ready* screen.
+
+## Platform specific band-aids
+
+Some platforms have mitigations in place where that small *200px \* 200px* UI
+will get scaled up by essentially **taking a screenshot of it and enlarging the image.**
+This will result in a blurry UI with diagonal and curved lines suffering the most.
+There is more hope with fonts where the vector information is still available to the platform,
+and instead of scaling up the image the text can be immediately drawn at the larger size.
+
+## A better solution
+
+The application should draw everything it can with **vector graphics**,
+and have **very large resolution image** assets available where vectors aren't viable.
+Then at runtime the application should identify the display pixel density
+and resize everything accordingly. The vector graphics are easy to resize and
+the large image assets would be scaled down to the size that makes sense for the specific display.
+
+## An even better way
+
+Druid aims to make all of this as **easy and automatic** as possible.
+Druid has expressive vector drawing capabilities that you should use whenever possible.
+Vector drawing is also used by the widgets that come included with druid.
+Handling different pixel densities is done at the `druid-shell` level already.
+In fact pixels mostly don't even enter the conversation at the `druid` level.
+The `druid` coordinate system is instead measured in **display points** (**dp**),
+e.g. you might say a widget has a width of **100dp**.
+These are conceptually the same as Microsoft's *device-independent pixels*,
+Google's *density-independent pixels*, and Apple's *points*.
```

Comment:
The CSS `px` (pixel) unit is probably worth putting in this list as well, the web being an extremely popular platform. (CSS doesn’t expose a device pixel unit.)

Reply:
The CSS pixel unit has quite a bit of baggage and has had several definitions over the years. However I guess it's fine to include it in the sense that newer developers probably are unaware of the older definitions anyway.